crates/services/txpool_v2/src/service.rs

```diff
@@ -411,7 +412,11 @@ where
                         .or_default();
                     block_height_expiration.push(tx_id);
                 }
-                self.shared_state.new_txs_notifier.send_replace(());
+                if executable {
+                    self.shared_state
+                        .new_executable_txs_notifier
+                        .send_replace(());
+                }
```

Comment:
It will not handle the case when we promote transaction from dependent to executed. I think we need to move `new_executable_txs_notifier` into the `TxPool` and notify in the case when we add a new executable transaction, and when we promote to the executable

Reply:
Done.